src/common/io.py

```diff
@@ -1657,55 +1697,62 @@ def set_options(self, options):
         self.options = options
 
 class ResultHandlerFile(ResultHandler):
-    """ 
-    Export an optimization or simulation result to file in Dymola's result file 
+    """
+    Export an optimization or simulation result to file in Dymola's result file
     format.
     """
     def __init__(self, model):
         self.model = model
-    
+
     def initialize_complete(self):
-        pass 
-    
+        pass
+
     def simulation_start(self):
         """
-        Opens the file and writes the header. This includes the information 
-        about the variables and a table determining the link between variables 
+        Opens the file and writes the header. This includes the information
+        about the variables and a table determining the link between variables
         and data.
         """
         opts = self.options
         model = self.model
-        
+
         #Internal values
         self.file_open = False
+        self._is_stream = False
         self.nbr_points = 0
-        
+
         self.file_name = opts["result_file_name"]
         try:
             self.parameters = opts["sensitivities"]
         except KeyError:
             self.parameters = None
-        
+
         if self.file_name == "":
             self.file_name=self.model.get_identifier() + '_result.txt'
-            
+
         #Store the continuous and discrete variables for result writing
         self.real_var_ref, self.int_var_ref, self.bool_var_ref = model.get_model_time_varying_value_references(filter=opts["filter"])
-        
+
         file_name = self.file_name
         parameters = self.parameters
-        
+
         # Open file
-        f = codecs.open(file_name,'w','utf-8')
+        if isinstance(self.file_name, str):
+            f = codecs.open(self.file_name,'w','utf-8')
+        else:
+            if not (hasattr(self.file_name, 'write') and hasattr(self.file_name, 'seek')):
+                raise fmi.FMUException("Instance set via option 'result_file_name' needs to support 'write' and 'seek' as required.")
```

Comment:
Same regarding the error message as above.

Reply:
Fixed